src/Kdyby/Console/DI/ConsoleExtension.php

```diff
@@ -82,18 +82,16 @@ public function loadConfiguration()
 		Nette\Utils\Validators::assert($config, 'array');
 		foreach ($config['commands'] as $i => $command) {
 			$def = $builder->addDefinition($this->prefix('command.' . $i));
-			list($def->factory) = Nette\DI\Compiler::filterArguments(array(
-				is_string($command) ? new Statement($command) : $command
-			));
-
-			if (class_exists($def->factory->entity)) {
-				$def->class = $def->factory->entity;
-			}
-
-			$def->setAutowired(FALSE);
-			$def->setInject(FALSE);
```

Comment:
You misunderstood me. I'm talking about runtime. If you run the app through console, Sf application takes over and loads all commands to ask them about their names and descriptions (at least) - this list is not cached and is fetched over and over in runtime. Which means the command has to be instantiated - which means all it's constructor dependencies have to be instantiaded, and if it has any inject properties or methods, that were wired in compile-time, they're now part of instantiating that service and will be also instantiated in memory. Which means all their dependencies are too. 

If you're following the rule to have light constructors, this shouldn't be a big deal for you, as it only creates "a few" (depending on the console command dependencies) services. But it has other problems, or you might be using some library that doesn't have exactly light constructors.

Having the `inject` disabled "by force" means that if you use inject properties (or methods) for the console commands, they will be resolved and injected on runtime. Which means a fraction of your object graph will be created. Unlike when all the commands use strictly constructor injection. 

That's a choice you can make by simply using a constructor or inject properties. Both will work. Both has advantages/disadvantages. It's up to the programmer to choose one.